fix: By Participant list for limited respondents (M2-8092)

farmerpaul opened the pull request:
<!-- Use this template as a guide to describe your pull request, and adjust as necessary. -->
<!-- Include information that helps your peers review your updates and understand this    -->
<!-- repository's history of changes over time.                                           -->

<!-- Delete any options that are not relevant -->

- [ ] ~~Tests for the changes have been added~~ Will to be done as part of M2&zwnj;-&zwnj;8024 (specific tests for this PR can't be added until general PDP tests have been added)

### 📝 Description

<!-- Contributions are welcome! If there is a corresponding      -->
<!-- JIRA ticket, link to it by replacing `#` with ticket number -->

🔗 [Jira Ticket M2-8092](https://mindlogger.atlassian.net/browse/M2-8092)

This PR adds support for viewing the PDP **By Participants** tab for a limited account, since they can be possible valid respondents if they had submissions done on their behalf via Take Now.

Also updated the copy for the limited account empty state on this tab to reflect this nuance.

> [!NOTE]
> Testing this PR requires being connected to the backend containing the changes in https://github.com/ChildMindInstitute/mindlogger-backend-refactor/pull/1654.

### 📸 Screenshots

#### With submissions:
https://github.com/user-attachments/assets/c68958a9-afbc-4bce-a794-1cbe3ecdb904

#### Without submissions:
![CleanShot 2024-11-07 at 14 46 50@2x](https://github.com/user-attachments/assets/8bc6adf2-6172-45a2-9a7c-e93bffe667f5)

### 🪤 Peer Testing

#### Preconditions:

An applet with at least one activity or flow, and a new limited account.

#### Steps:

1. Navigate to the PDP for the limited account, then the **By Participant** tab.
    **Expected outcome:** The empty state displays the [updated copy](https://www.figma.com/design/6BNx63peNVzrHAF1FoZ78x/Multi-Informant-2024?node-id=13935-119943&t=ubTxBcV95pvhJbap-4).
1. Navigate to the **Applet > Activities** tab.
2. For any activity or flow, click **Take Now** from the ⋯ menu, providing the limited account for the **Who will be providing the responses?** dropdown, and any participants for the **Who will be inputting the responses?** and **Who are the responses about?** dropdowns.
3. Complete the assessment in the Web App so that there is at least one submission by the limited account as the respondent.
4. Navigate to the PDP for the limited account, then the **By Participant** tab.
    **Expected outcome:** The activity or flow just completed should be listed in `Inactive` state.
6. Expand the card for the activity or flow.
    **Expected outcome:** One subject entry should be listed for the assessment just completed via Take Now.
7. Open the ⋯ menu for that activity or flow, and hover over the **Assign** menu item.
    **Expected outcome:** A tooltip explaining why it's disabled should be shown.
6. Click the **Take Now** menu item.
    **Expected outcome:**
    The Take Now dialogue should be pre-populated with the limited account in the **Who will be providing the responses?** dropdown, and the participant shown in the row that you clicked the ⋯ menu item from in the **Who are the responses about?** dropdown. The checkbox should be disabled and unchecked, and the **Who will be inputting the responses?** dropdown should be empty.

farmerpaul committed:
fix: By Participant list for limited respondents

Treat limited account as possible valid respondents, since they could
have had submissions done on their behalf via Take Now.

farmerpaul committed:
fix: update empty state copy for limited accounts

sultanofcardio commented on src/modules/Dashboard/features/Participant/Assignments/AssignmentsTab/AssignmentsTab.hooks.tsx:
```diff
@@ -207,7 +208,7 @@ export const useAssignmentsTab = ({
       const isAssignDisplayed =
         canAssign && (!autoAssign || status === ActivityAssignmentStatus.Hidden);
-      const isAssignDisabled = !isAssignable || isTargetTeamMember;
+      const isAssignDisabled = !isAssignable || isTargetTeamMember || isLimitedRespondent;
```
It's currently possible to access the assign menu item from the participants table more menu on a limited account. When this happens, the participant is pre-filled as the target of the assignment.

<img width="360" alt="image" src="https://github.com/user-attachments/assets/0664ba2d-b0b5-49d9-baf0-74adb585ae43"> <img width="360" alt="image" src="https://github.com/user-attachments/assets/db63e116-7880-427d-901d-67ef5183ad3b">


I think it would make sense to take a similar approach on at least the "about participant" view of the assignments tab. Though if we just did the same thing for both views, that'd also be fine.

farmerpaul replied:
Good catch. That's actually a regression caused by incorrect logic for determining if the respondent is a limited account. What you're describing is actually part of the original AC for [M2-7828](https://mindlogger.atlassian.net/browse/M2-7828).

In fixing that bug I noticed another bug that would mislead users into thinking activities/flows could hypothetically be assigned to a limited account as a respondent; see explanation in [this commit](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1974/commits/1221808051864316a18243ea245b401fa6b5a378).

sultanofcardio commented on src/modules/Dashboard/features/Participant/Assignments/EmptyState/EmptyState.types.ts:
```diff
@@ -4,4 +4,6 @@ export type EmptyStateProps = {
   title: string;
   onClickAssign?: () => void;
+  /** @default '50.7rem' */
+  width?: string;
```
`maxWidth`?

```suggestion
  maxWidth?: string;
```

sultanofcardio reviewed (approved):
Works as expected. Approving with a couple suggestions

farmerpaul committed:
fix: assign logic for limited respondents

Ensured the `isLimitedRespondent` is only true if a respondent subject
is even present. This ensures assign/unassign menu items follow correct
logic on the About Participant tab.

Also, for the By Participant tab, checks first if the respondent is a
limited account before hiding the Assign menu item and showing the
Unassign menu item, so that when an activity or flow is set to
auto-assign, the user isn't misled into thinking that auto-assigned
activities/flows could have been assigned to the limited account as the
respondent.

farmerpaul committed:
fix: `width` ⇒ `maxWidth`
